### PR TITLE
Use error overlay for errors (fixes #395)

### LINKED ISSUE #395: Hook up error handling
The error overlay is available on the addon. It needs hooking up to handle error states.

### PR DESCRIPTION
**Pull request by mstriemer**
I had a little animation on the overlay at first but getting it to animate out was tricky so I removed it. It appears quite abruptly so we should probably add something but I'll let @pwalm say what. Maybe worth noting that I barely missed the close button there and it didn't register my click. I guess we should make that clickier.

error-overlay-nicer mov
> No animation

error-overlay-animate-in mov
> In animation

Fixes #395.

### CODE REVIEW

File: tests/client/helpers.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ export function unexpectedSuccess() {
  */
 export function getFakeI18nInst() {
   return {
-    gettext: sinon.stub(),
+    gettext: sinon.spy((str) => str),
```
Comment (muffinresearch) [inline tests/client/helpers.js]: heh, nice.

File: src/disco/components/InstallButton.js [hunk 2]
```diff
@@ -38,11 +38,13 @@ export class InstallButton extends React.Component {
...
     e.preventDefault();
-    const { guid, install, installURL, name, status, installTheme, type, uninstall } = this.props;
+    const {
+      guid, i18n, install, installURL, name, status, installTheme, type, uninstall,
+    } = this.props;
     if (type === THEME_TYPE && status === UNINSTALLED) {
       installTheme(this.refs.themeData, guid, name);
     } else if (status === UNINSTALLED) {
-      install({ guid, installURL, name });
+      install({ guid, i18n, installURL, name });
```
Comment (muffinresearch) [inline src/disco/components/InstallButton.js]: Hmm, I guess you have to pass i18n down into these functions?

File: tests/client/disco/components/TestAddon.js [hunk 3]
```diff
@@ -220,16 +227,40 @@ describe('<Addon />', () => {
...
+
+    it('sets status to error on onDownloadFailed', () => {
+      const dispatch = sinon.spy();
+      const guid = '{my-addon}';
+      const i18n = getFakeI18nInst();
+      const handler = makeProgressHandler(dispatch, guid, i18n);
+      handler({state: 'STATE_SOMETHING'}, {type: 'onDownloadFailed'});
+      assert(dispatch.calledWith({
+        type: 'INSTALL_ERROR',
+        payload: {guid, errorMessage: 'Download failed. Please check your connection.'},
```
Comment (muffinresearch) [inline tests/client/disco/components/TestAddon.js]: I suppose an alternative to passing `errorMessage` around would be to pass a constant and key a translation off the constant closer to the error message display. That would avoid all the passing i18n around.

**Comment by muffinresearch** [comment]
r+wc

**Comment by mstriemer** [comment]
I put 5px of padding on the close button so it's a bit easier to click. Still no animation on the overlay appearing, will wait for feedback from @pwalm first.

**Comment by pwalm** [comment]
BRING ON THE ANIMATION! That looks sweet. But how about when the user turns on the switch, it turns back off (due to whatever error) and then the error animates in. That way the user will see that the switch didn't work before the error pops up.

**Comment by pwalm** [comment]
Oh and good call on the 5px of padding 👍

**Pull request merged**